services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -3257,45 +3424,180 @@ public Optional<Schema> getReplicationMetadataSchema(
         throw new VeniceException("Store: " + storeName + " is not initialized with a version yet");
       }
 
-      /**
-       * Don't use {@link Store#getCurrentVersion()} here since it is always 0 in parent controller
-       */
-      Version version = versions.get(versions.size() - 1);
-      if (version.getStatus() == ERROR) {
+      Version hybridVersion = null;
+      versions.sort(Comparator.comparingInt(Version::getNumber).reversed());
+      for (Version version: versions) {
+        if (version.getHybridStoreConfig() != null && version.getStatus() == ONLINE) {
+          hybridVersion = version;
+          break;
+        }
+      }
+
+      if (hybridVersion == null) {
+        LOGGER.error(
+            "Could not find an online hybrid store version for incremental push: {} on store: {} in cluster: {}",
+            pushJobId,
+            storeName,
+            clusterName);
         throw new VeniceException(
-            "cannot have incremental push because current version is in error status. " + "Version: "
-                + version.getNumber() + " Store:" + storeName);
+            "No ONLINE hybrid store version found for store: " + storeName + " in cluster: " + clusterName);
+      }
+      LOGGER.info(
+          "Found hybrid version: {} for store: {} in cluster: {}. Will use it as a reference for incremental push: {}",
+          hybridVersion.getNumber(),
+          storeName,
+          clusterName,
+          pushJobId);
+
+      // If real-time topic is not required, no need to check for its presence
+      if (!isRealTimeTopicRequired(store, hybridVersion)) {
+        return hybridVersion;
       }
 
       PubSubTopic rtTopic = pubSubTopicRepository.getTopic(Utils.getRealTimeTopicName(store));
       if (!getTopicManager().containsTopicAndAllPartitionsAreOnline(rtTopic) || isTopicTruncated(rtTopic.getName())) {
+        LOGGER.error(
+            "Incremental push: {} cannot be started for store: {} in cluster: {} because the topic: {} is either absent or being truncated",
+            pushJobId,
+            storeName,
+            clusterName,
+            rtTopic);
         resources.getVeniceAdminStats().recordUnexpectedTopicAbsenceCount();
         throw new VeniceException(
             "Incremental push cannot be started for store: " + storeName + " in cluster: " + clusterName
                 + " because the topic: " + rtTopic + " is either absent or being truncated");
       }
-      return version;
+
+      if (hybridVersion.isSeparateRealTimeTopicEnabled()) {
+        PubSubTopic separateRtTopic = pubSubTopicRepository.getTopic(Version.composeSeparateRealTimeTopic(storeName));
+        if (!getTopicManager().containsTopicAndAllPartitionsAreOnline(separateRtTopic)
+            || isTopicTruncated(separateRtTopic.getName())) {
+          LOGGER.error(
+              "Incremental push: {} cannot be started for store: {} in cluster: {} because the topic: {} is either absent or being truncated",
```

Comment:
Just to clarify that by implementation, inc push is only pushed to one of the above topics based on VPJ config. But I think more checking is always good. nit: can consider wrapping the lines into a small check method and reuse for RT and sepRT.